src/chatd.cpp

```diff
@@ -3840,6 +3866,10 @@ Idx Chat::msgConfirm(Id msgxid, Id msgid, uint32_t timestamp)
     }
 
     CALL_DB(addMsgToHistory, *msg, idx);
+    if (mOldestIdxInDb == CHATD_IDX_INVALID)
+    {
+        mOldestIdxInDb = mDbInterface->getOldestIdx();
```

Comment:
At this point, the history was empty and now it has one message that the client sent and just got confirmed, right?
In that case, you don't need a DB query:
```suggestion
        mOldestIdxInDb = idx;
```